File: solutions/javascript/allergies/1/allergies.js
//
// This is only a SKELETON file for the 'Allergies' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Allergies {
  static list = ['eggs', 'peanuts', 'shellfish', 'strawberries', 'tomatoes', 'chocolate', 'pollen', 'cats']
  constructor(number) {
    this.number = number
  }

  list() {
    return Allergies.list.filter(item => this.allergicTo(item));
  }

  allergicTo(alergen) {
    let test = 2 ** Allergies.list.indexOf(alergen)
    return (test & this.number) === test;
  }
}

File: solutions/nim/allergies/1/allergies.nim
type
  Allergents* = enum
    Eggs, Peanuts, Shellfish, Strawberries, Tomatoes, Chocolate, Pollen, Cats
proc isAllergicTo*(score: int,allergy: Allergents):bool =
  return (score and (1 shl int(allergy))) != 0
proc allergies*(score: int): set[Allergents] =
  for i in ord(low(Allergents))..ord(high(Allergents)):
    if isAllergicTo(score, Allergents(i)):
      result = result + {Allergents(i)}
